interop/xds_federation/client.go

```diff
@@ -122,7 +122,8 @@ func main() {
 		go func(c clientConfig) {
 			overallDeadline := time.Now().Add(time.Duration(*soakOverallTimeoutSeconds) * time.Second)
 			ctxWithDeadline, cancel := context.WithDeadline(ctx, overallDeadline)
```

Comment:
Similar: it's best to `defer cancel()` here instead (nearly always).

The main time you wouldn't `defer` is if the context is created inside a loop, because you don't want to stack up a very large number of `defer`s.